openverse-api/catalog/urls/__init__.py

```diff
@@ -191,147 +186,88 @@
-        method='post',
-        responses={
-            "201": openapi.Response(
-                description="OK",
-                examples=images_report_create_201_example,
-                schema=ReportImageSerializer
-            )
-        },
-        request_body=report_image_request,
-        code_examples=[
-            {
-                'lang': 'Bash',
-                'source': report_image_bash
-            }
-        ]
-    )(ReportImageView.as_view())
+discontinuation_message = {
+    'error': 'Gone',
+    'reason': 'This API endpoint has been discontinued.'
+}
 
 versioned_paths = [
-    path('', schema_view.with_ui('redoc', cache_timeout=None), name='root'),
-    path('auth_tokens/register', Register.as_view(), name='register'),
     path('rate_limit', CheckRates.as_view(), name='key_info'),
-    path(
-        'auth_tokens/verify/<str:code>',
-        VerifyEmail.as_view(),
-        name='verify-email'
-    ),
-    re_path(
-        r'auth_tokens/',
-        include('oauth2_provider.urls', namespace='oauth2_provider')
-    ),
+    path('auth_tokens/', include(auth_tokens_patterns)),
 
-    path(
-        'audio/stats',
-        AudioStats.as_view(),
-        name='audio-stats'
-    ),
-    path(
-        'audio/<str:identifier>',
-        AudioDetail.as_view(),
-        name='audio-detail'
-    ),
-    re_path('audio', SearchAudio.as_view(), name='audio'),
-    path(
-        'recommendations/audio/<str:identifier>',
-        RelatedAudio.as_view(),
-        name='related-audio'
-    ),
+    # Audio
+    path('audio/', include(audio_patterns)),
+
+    # Images
+    path('images/', include(images_patterns)),
 
+    # Deprecated
     path(
-        'images/stats',
-        ImageStats.as_view(),
-        name='image-stats'
+        'sources',
+        RedirectView.as_view(pattern_name='image-stats', permanent=True),
+        name='about-image'
     ),
     path(
-        'images/<str:identifier>',
-        ImageDetail.as_view(),
-        name='image-detail'
+        'recommendations/images/<str:identifier>',
+        RedirectView.as_view(pattern_name='image-related', permanent=True),
+        name='related-images',
     ),
     path(
-        'images/<str:identifier>/report',
-        decorated_report_image_view,
-        name='report-image'
+        'oembed',
+        RedirectView.as_view(
+            pattern_name='image-oembed',
+            query_string=True,
+            permanent=True
+        ),
+        name='oembed'
     ),
     path(
-        'recommendations/images/<str:identifier>',
-        RelatedImage.as_view(),
-        name='related-images'
+        'thumbs/<str:identifier>',
+        RedirectView.as_view(pattern_name='image-thumb', permanent=True),
+        name='thumbs'
     ),
-    re_path('images', SearchImages.as_view(), name='images'),
-    path(  # Deprecated
-        'sources',
-        RedirectView.as_view(pattern_name='image-stats', permanent=True),
-        name='about-image'
+
+    # Discontinued
```

Comment:
Is there a difference between a `deprecated` and a `discontinued` url?

Reply:
Deprecated URLs will be redirected to new paths, discontinued ones will return a 4xx error message.